### PR TITLE
[14.0][FIX] account_financial_report: Trial Balance with 'hide account at 0'

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT34490 TT35127

Fix 'Trial Balance' report: The 'Trial Balance' report shows accounts at 0 even if 'Hide accounts at 0' option is checked.

### CODE REVIEW

**Comment by ernestotejeda** [comment]
ping @pedrobaeza @sergio-teruel @carlosdauden  @JoanSForgeFlow

**Comment by ernestotejeda** [comment]
ping @pedrobaeza

File: account_financial_report/report/trial_balance.py [hunk 1]
```diff
@@ -300,6 +300,29 @@ def _compute_partner_amount(
                         ] += round(tb["amount_currency"], 2)
         return total_amount, partners_data
 
+    def _remove_accounts_at_cero(self, total_amount, show_partner_details):
+        def is_removable(d):
+            return (
+                d["initial_balance"] == 0
```
Comment (pedrobaeza) [inline account_financial_report/report/trial_balance.py]: This should be used with `float_is_zero` (and if not possible to get decimal precision, put 8 digits).

File: account_financial_report/report/trial_balance.py [hunk 2]
```diff
@@ -300,6 +300,29 @@ def _compute_partner_amount(
...
+            )
+
+        accounts_to_remove = []
+        for acc_id, ta_data in total_amount.items():
+            if is_removable(ta_data):
+                accounts_to_remove.append(acc_id)
+            elif show_partner_details:
+                partner_to_remove = []
+                for key, value in ta_data.items():
+                    if isinstance(key, int) and is_removable(value):
```
Comment (pedrobaeza) [inline account_financial_report/report/trial_balance.py:324]: Why do you need to do this `isinstance` thing? Always explain as comment the weird things.

Comment (ernestotejeda) [reply account_financial_report/report/trial_balance.py:324]: I have put a comment in the code, although it is difficult for me to explain. The image below shows the value of the total_amount variable (the main dict used for printing the report) while debugging the code in a use case. If the show_partner_details option is checked, in addition to the "initial_balance" ... "ending_balance" keys (marked in red), the same account data dictionary will also have the partner IDs as key (marked in blue) with their respective data. That's why i used isinstance method. Check the comment i have put.

image

And this is the result with this data and without my code:

image

**Review by Shide — APPROVED** [review]

**Comment by Shide** [comment]
@pedrobaeza are you ok with the new changes?

**Review by rafaelbn — APPROVED** [review]
Tested functionally 👍 

https://github.com/OCA/account-financial-reporting/blob/14.0/account_financial_report/wizard/trial_balance_wizard.py#L55-L57

```
    hide_account_at_0 = fields.Boolean(
        string="Hide accounts at 0",
        default=True,
        help="When this option is enabled, the trial balance will "
        "not display accounts that have initial balance = "
        "debit = credit = end balance = 0",
    )
```

**Comment by Shide** [comment]
@pedrobaeza are you ok with the changes (comments) of @ernestotejeda ?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
It's not working properly yet.

**Comment by ernestotejeda** [comment]
@pedrobaeza could please you check again with this new commit?

**Comment by pedrobaeza** [comment]
@ernestotejeda can you remove the "computed hierarchy" option as now the hierarchy is already computed in the records and no need to compute them on the fly?

**Comment by ernestotejeda** [comment]
> @ernestotejeda can you remove the "computed hierarchy" option as now the hierarchy is already computed in the records and no need to compute them on the fly?

@pedrobaeza change done

**Comment by ernestotejeda** [comment]
Travis is red, i will check it tomorrow

**Comment by ernestotejeda** [comment]
@pedrobaeza travis is green now

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
/ocabot merge minor

Please fw-port it to v15

**Pull request merged**